rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/aws/AWSBakeHandler.groovy

```diff
@@ -231,14 +231,21 @@ public class AWSBakeHandler extends CloudProviderBakeHandler {
     return new Bake(id: bakeId, ami: amiId, image_name: imageName, artifacts: artifacts)
   }
 
-  private String lookupAmiByName(String name, String region, String account, VmType vmType) {
+  private String lookupAmiByName(String name, String region, String account, VmType vmType, boolean mostRecent) {
     def images = AuthenticatedRequest.allowAnonymous(
       {
         retrySupport.retry({
           clouddriverService.findAmazonImageByName(name, account, region)
         }, 3, Duration.ofSeconds(3), false)
       }
     )
+
+    if (mostRecent) {
+      def matchingImages = images?.findAll { it?.attributes?.virtualizationType == vmType }
+      def image = matchingImages?.max {it?.attributes?.creationDate }
+      return image?.amis?.get(region)?.first()
```

Comment:
Given that line 246 and 250 are the same, would it be beneficial to reduce this to one return statement?

Reply:
Good idea! Fixed that up